source/develop/release-management/features/storage/data-path-operations.html.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Data Path operations on any host
+category: feature
+authors: alitke, nsoffer, laravot, frolland
+feature_name: Data Path Operations On Any Host
+feature_modules: storage,engine,vdsm
+feature_status: WIP 4.1
+---
+
+# Data Path Operations On Any Host
+
+## Summary
+
+Until now almost all data path operations on the hosts were performed only on the elected SPM, causing a potential bottleneck.
+This feature enables data path operations on any host. 
+
+### Owner
+
+*   Adam Litke (<alitke@redhat.com>)
+*   Liron Aravot (<laravot@redhat.com>)
+*   Nir Soffer (<nsoffer@redhat.com>)
+*   Fred Rolland (<frolland@redhat.com>)
+
+### Detailed Description
+
+There are a lot of data path operations in oVirt. For example: move disk, create VM from template, create template from VM, live storage migration and more.
+These operations are performed on the SPM host, potentially creating a bottleneck in scale setups or if the operations are performed frequently.
+
+By adding a new verb 'SDM.copy_data' available on all hosts, the engine is capable to convert the data path operations flows to run on any host.
+
+### Benefit to oVirt
+
+With this feature the load of the data operations is spread between all the hosts, preventing the throttling of the SPM host. 
+
+### User Experience
+
+A progress indication of the operation is available on the disks status for the "Move Disk" operation.
+Otherwise, there are no changes in the user experience.
+
+
+### High level flow
+
+Here is the high level flow for "Move Disk" operation
+- User starts disk move operation via UI or REST, no change is needed
+- Engine clone image structure via the SPM host and existing VDSM apis
+- Engine select hosts for copying volume
+- Engine starts copy_data operation on selected hosts
+- On each host, VDSM schedule copy_data job, copying data from source volume to destination volume
+- Engine monitor jobs progress
+- Engine wait until all jobs are done
+
+For "Create VM from Template (Clone/Independent)" operation, the flow is the same as above.
+The only difference is that Templates have only one volume.
+
+### VDSM
+
+The following verb is added:
+
+```
+SDM.copy_data:
+    added: '4.1'
+    description: Copy data from one volume to another.
+    params:
+    -   description: A UUID to be used for tracking the job progress
+        name: job_id
+        type: *UUID
+
+    -   description: The source endpoint
+        name: source
+        type: *CopyDataEndpoint
+
+    -   description: The destination endpoint
+        name: destination
+        type: *CopyDataEndpoint
+```
+
+### Engine
+
+The engine will monitor the operation using the `Job` API.
+In case the job is not present, the status of the operation will determined by the lease status on the volume information retrieved from VDSM.
```

Comment:
We examine the lease state to determine if a copy may still be active.  If the lease is free we can know whether the copy completed by checking the volume legality.